sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -31,11 +31,22 @@ object InterpretedPredicate {
   def create(expression: Expression, inputSchema: Seq[Attribute]): InterpretedPredicate =
     create(BindReferences.bindReference(expression, inputSchema))
 
-  def create(expression: Expression): InterpretedPredicate = new InterpretedPredicate(expression)
+  def create(expression: Expression): InterpretedPredicate = {
+    val predicate = new InterpretedPredicate(expression)
+    predicate.initialize(0)
```

Comment:
The consideration then was the consistency of class InterpretedPredicate, so add initialize method of the class InterpretedPredicate.